### PR TITLE
pre-commit: match openpilot mypy config

### PR DESCRIPTION
**Pull request by GregorKikelj**
PR for https://github.com/commaai/openpilot/pull/24287

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Make sure to enable the same flags here.

**Comment by GregorKikelj** [comment]
Can you enable workflows btw? Want to ensure it all works :)

File: requirements_examples.txt [hunk 1]
```diff
@@ -9,4 +9,5 @@ matplotlib
 cffi
 gmplot
 pycurl
+types-pycurl
```
Comment (adeebshihadeh) [inline requirements_examples.txt:12]: this goes in the pre-commit config. see the openpilot pre-commit config for an example

Comment (GregorKikelj) [reply requirements_examples.txt]: Thanks

**Comment by GregorKikelj** [comment]
Could add the `--warn-unused-ignores` to laika but I get a lot of missing stubs even after fixing numpy and pycurl
```
downloader.py:3: error: Cannot find implementation or library stub for module named "hatanaka"
raw_gnss.py:1: error: Skipping analyzing "scipy.optimize": module is installed, but missing library stubs or py.typed marker
raw_gnss.py:1: error: Skipping analyzing "scipy": module is installed, but missing library stubs or py.typed marker
dgps.py:4: error: Skipping analyzing "scipy.spatial": module is installed, but missing library stubs or py.typed marker
dgps.py:4: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
```

**Pull request merged**